optuna/pruners/base.py

```diff
@@ -1,29 +1,31 @@
 import abc
 import six
 
-from optuna.storages import BaseStorage  # NOQA
+from optuna.types import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from optuna import Study  # NOQA
+    from optuna.structs import FrozenTrial  # NOQA
 
 
 @six.add_metaclass(abc.ABCMeta)
 class BasePruner(object):
     """Base class for pruners."""
 
     @abc.abstractmethod
-    def prune(self, storage, study_id, trial_id, step):
-        # type: (BaseStorage, int, int, int) -> bool
+    def prune(self, study, trial, step):
+        # type: (Study, FrozenTrial, int) -> bool
```

Comment:
Use Study object, instead of InTrialStudy (See https://github.com/pfnet/optuna/pull/491).